workbench-core/accounts/src/utilities/hostingAccountLifecycleService.ts

```diff
@@ -71,6 +72,46 @@ export default class HostingAccountLifecycleService {
     return this._accountService.getAccount(accountId, includeMetadata);
   }
 
+  /**
+   * Create/Upload template and return its URL
+   * @param awsAcctId - ID of account to retrieve
+   *
+   * @returns A URL to a prepopulated template for onboarding the hosting account.
+   */
+  public async buildTemplateURLForAccount(awsAcctId: string, externalId: string): Promise<TemplateResponse> {
+    // Share the artifacts bucket with the new hosting account
+
+    const {
+      [process.env.ACCT_HANDLER_ARN_OUTPUT_KEY!]: accountHandlerRoleArn,
+      [process.env.STATUS_HANDLER_ARN_OUTPUT_KEY!]: statusHandlerRoleArn,
+      [process.env.API_HANDLER_ARN_OUTPUT_KEY!]: apiHandlerRoleArn,
+      [process.env.S3_ARTIFACT_BUCKET_ARN_OUTPUT_KEY!]: artifactBucketArn,
+    } = await this._aws.helpers.cloudformation.getCfnOutput(this._stackName, [
+      process.env.ACCT_HANDLER_ARN_OUTPUT_KEY!,
+      process.env.STATUS_HANDLER_ARN_OUTPUT_KEY!,
+      process.env.API_HANDLER_ARN_OUTPUT_KEY!,
+      process.env.S3_ARTIFACT_BUCKET_ARN_OUTPUT_KEY!,
+    ]);
+
+    const templateParameters: AccountCfnTemplateParameters = {
+      accountHandlerRole: accountHandlerRoleArn,
+      apiHandlerRole: apiHandlerRoleArn,
+      enableFlowLogs: 'true',
+      externalId: externalId,
+      launchConstraintPolicyPrefix: '*', // We can do better, get from stack outputs?
+      launchConstraintRolePrefix: '*', // We can do better, get from stack outputs?
+      mainAccountId: process.env.MAIN_ACCT_ID!,
+      namespace: process.env.STACK_NAME!,
+      stackName: process.env.STACK_NAME!.concat('-hosting-account'),
+      statusHandlerRole: statusHandlerRoleArn
+    };
+
+    const s3Client = new S3Client({
```

Comment:
In our code we don't initialize and AWS SDK. We normally just pull it from `awsService`. Can we move this implementation [here](https://github.com/aws-solutions/solution-spark-on-aws/blob/b045c4c0d21724d13ae7e6e92c6dda000e6e7dc6/workbench-core/base/src/aws/helpers/s3Service.ts). 

Then we can do `this._aws.helpers.s3.generatePresignedUrl(s3Bucket, s3Key)`. 

I think this code can live in the method above
```
const command = new GetObjectCommand( {
      Bucket: bucket,
      Key:key,
    });
    const signedUrl = await getSignedUrl(s3Client, command, { expiresIn: 15 * 60 });
 ```